tests/utils.go

```diff
@@ -84,6 +84,7 @@ func NewRandomReplicaSetFromVMI(vmi *v1.VirtualMachineInstance, replicas int32)
 	return rs
 }
 
+// Deprecated: GetRunningVirtualMachineInstanceDomainXML should not be used by end-to-end tests
```

Comment:
Verifying we generate proper libvirt XML should indeed happen in unit tests.
Verifying libvirt populated the right defaults and whatnot should happen in libvirt.
However, verifying we properly updated the XML after a hotplug for example, is something that I think belongs to KubeVirt e2e... WDYT?

Reply:
We could ensure the hotplug actually happened (new disk available or so) instead of relying on the libvirt change
This may not be trivial in some cases but it would make this a more honest "end to end" test

Reply:
+ a unit test for the function that updates the XML during hotplug.

Reply:
> + a unit test for the function that updates the XML during hotplug.

That would be libvirt

Reply:
> > \+ a unit test for the function that updates the XML during hotplug.
> 
> That would be libvirt

Ah, so no point for a kubevirt-side unit test, similar to

> Verifying libvirt populated the right defaults and whatnot should happen in libvirt.

Reply:
I think we would like to have a integration tests with libvirt. For that I think it could be enough to have launcher running and the whole e2e env would not be needed. This is possible future improvement.

Reply:
I concur with Jed. The domain XML of libvirt is their published API. Ensuring their API works is the domain of libvirt. Ensuring we honor the API is our domain.

Why are we conflating functional and integration tests?

Reply:
To be clear, I'm not saying there's no value in an integration test. I'm suggesting this might not be where that gets done.

Reply:
> Ensuring their API works is the domain of libvirt

In an ideal world, we would not have to test that the API works, but seeing as we're the bridge between libvirt (or, others in the future) and container runtimes, I think we can't unfortunately escape having to test the end result, at least to some degree.